.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,141 @@
+          if [ "${{ matrix.source }}" == "upstream" ]; then
+            # quay.io/buildah/upstream:master
+            echo ::set-output name=buildah_tag::"${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:master"
+          fi
+
+          if [ "${{ matrix.source }}" == "stable" ]; then
+            VERSION=v"$(echo "${{ steps.sniff_test.outputs.version_output }}" | head -n 1 | awk '{print $2}')"
+            # quay.io/buildah/stable:vX.X.X
+            echo ::set-output name=buildah_tag::"${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:${VERSION}"
+            # quay.io/contaners/buildah:vX.X.X
+            echo ::set-output name=containers_tag::"${{ env.CONTAINERS_QUAY_REGISTRY}}/buildah:${VERSION}"
+          fi
+
+          # have labels environment variable to use it in the further steps.
+          # Env variable is used to store multi-line value
+          echo "LABELS<<EOF" >> $GITHUB_ENV
+          echo "org.opencontainers.image.source=${{ github.event.repository.html_url }}" >> $GITHUB_ENV
+          echo "org.opencontainers.image.revision=${{ github.sha }}" >> $GITHUB_ENV
+          echo "org.opencontainers.image.created=$(date -u --iso-8601=seconds)" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      # Separate steps to login and push for buildah and containers quay
+      # repositories are required, because 2 sets of credentials are used and `docker
+      # login` as well as `podman login` do not support having 2 different
+      # credential sets for 1 registry.
+      # At the same time reuse of non-shell steps is not supported by Github Actions via
+      # anchors or composite actions
+
+      # Push to buildah Quay repo stable and upstream Buildah
+      - name: Login to buildah Quay registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.BUILDAH_QUAY_REGISTRY }}
+          username: ${{ secrets.BUILDAH_QUAY_USERNAME }}
+          password: ${{ secrets.BUILDAH_QUAY_PASSWORD }}
+
+      - name: Push images to buildah Quay
+        uses: docker/build-push-action@v2
+        with:
+          cache-from: type=registry,ref=localhost:5000/buildah/${{ matrix.source }}
+          cache-to: type=inline
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: ${{ steps.image_info.outputs.buildah_tag }}
+          labels: |
+            ${{ env.LABELS }}
+
+      # Push to containers Quay repo only stable Buildah
+      - name: Login to containers Quay registry
+        if: ${{ matrix.source == 'stable' }}
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.CONTAINERS_QUAY_REGISTRY}}
+          username: ${{ secrets.CONTAINERS_QUAY_USERNAME }}
+          password: ${{ secrets.CONTAINERS_QUAY_PASSWORD }}
+
+      - name: Push images to containers Quay
+        if: ${{ matrix.source == 'stable' }}
```

Comment:
I wonder if might be more clear, if this was:

```
    if: ${{ steps.image_info.outputs.containers_tag }} != ""
```

what do you think?

Reply:
btw, I can see master tag at https://quay.io/repository/containers/buildah?tag=latest&tab=tags . Can we replace the current build setup with multi-arch one? If the source of that build is contrib/buildahimage/upstream, of course.
Then no check is required at all.

Reply:
I think so?  The source of that is a cronjob on @TomSweeneyRedHat laptop :grinning:   I already poked him about this PR, so he's aware.  My understanding is this PR will improve the situation...though Tom will tell us if it's a no-go on the 'master' tag.